scripts/gen_release_notes/git_parser/git_parser.py

```diff
@@ -0,0 +1,270 @@
+"""
+This module contains all of the methods used to interact with git
+"""
+
+import os
+
+import generate
+import requests
+from helper import helper
+
+# Secrets
+# These are set via ENV variables
+pat = os.getenv("GITHUB_TOKEN", "")
+
+pr_numb_regex = r"^#[0-9]+$"
+merge_commit_regex = r"^Merge"
+cherry_commit_regex = r"\(cherry picked from commit \w+\)"
+
+loading_icons = ["|", "/", "-", "\\"]
+
+
+def get_all_changed_files(repo_owner, repo_name, pr_number):
+    """
+    This function grabs all of the changed files that are in a given PR, given the following:
+    - repo owner
+    - repo name
+    - pr number
+
+    Returns a list of all changed files. If it cannot find any,
+    """
+
+    payload = {"Authorization": "token {secret}".format(secret=pat)}
```

Comment:
Would it make more sense if the script just errors out if the `GITHUB_TOKEN` isn't defined (line 13 implies this)?

Alternatively - if it's possible to call this API without authentication (unlikely), then this probably shouldn't be part of the payload.

Reply:
Good point; this script requires that value to be here, so it would make sense to just fail fast if it can't find that token.